sdk/identity/identity/samples/AzureIdentityExamples.md

```diff
@@ -417,7 +439,7 @@ else, if the Identity provider of your Azure Stack is Active Directory Federatio
 The following example demonstrates authenticating a `SecretClient` from the [@azure/keyvault-secrets][secrets_client_library] against an Azure Key Vault hosted in Azure Stack.
 
 ```ts
-function main() {
+async function main() {
```

Comment:
That is correct! However, most of the methods of the class instances we're using in this main function are asynchronous. If users copied this code and expanded it to use these class instances, they may end up having errors because of the lack of an asynchronous context 🤔

Reply:
This is definitely a different format than all other code samples in this file, it would be preferable if we are consistent.

How about using the current format of creating a function named `with...`, and make it return the client?

Reply:
Do you mean something like `withSecretClient`? Seems oddly specific. I’ll remove the async in the mean time.